client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useDraftOutboundLines.tsx

```diff
@@ -32,11 +32,12 @@ export const useDraftOutboundLines = (
   const [draftStockOutLines, setDraftStockOutLines] = useState<
     DraftStockOutLine[]
   >([]);
+  const noStockLines = !data?.nodes.length;
 
   useConfirmOnLeaving(isDirty);
 
   useEffect(() => {
-    if (!item) {
+    if (!item || noStockLines) {
```

Comment:
Just wondering if this is necessary. It still works fine without this check, as the subsequent `rows.map...` has nothing in it so ends up basically the same.

Was there a particular reason for adding this check?

Reply:
It adds an empty stockOutLine to the array, so the messages don't show up...